includes/Preferences.php

```diff
@@ -139,12 +139,19 @@ static function getUserPreference( $name, $info, $user ) {
 		return $val;
 	}
 
-	private static function getUserPreferenceHelper(User $user, $property) {
-		if (in_array($property, self::getAttributes())) {
-			return $user->getGlobalAttribute($property);
-		} else {
-			return $user->getGlobalPreference($property);
+	private static function getUserPropertyHelper( User $user, $property ) {
+		$scopeService = self::preferenceScope();
+
+		if ( in_array( $property, self::getAttributes() ) ) {
+			return $user->getGlobalAttribute( $property );
+		} elseif ( $scopeService->isGlobalPreference( $property ) ) {
+			return $user->getGlobalPreference( $property );
+		} elseif ( $scopeService->isLocalPreference( $property ) ) {
+			list( $prefName, $wikiId ) = $scopeService->splitLocalPreference( $property );
+			return $user->getLocalPreference( $prefName, $wikiId );
 		}
+
+		return null;
```

Comment:
When doing this I noticed some non-attributes and non-preferences got here, specifically:

```
emailauthentication
username
usergroups
emailaddress
password
oldsig
fbconnect-disconnect
adoptionmails-label-<some_number>
```

but I did a search on the database and none of them has any entries, so they're always returning `null` for any user, except `username` which is something that I spoke to @jsutterfield about, and it seems ok for this to return `null` in that case as well.